python/test/unit/language/test_tlx.py

```diff
@@ -70,34 +70,66 @@ def dual_add(x, y, a, b):
     reason="Requires compute capability >= 9 for NV",
 )
 @pytest.mark.parametrize("BLOCK_SIZE", [(1024)])
-def test_alloc_barriers(BLOCK_SIZE, device):
+def test_mbarriers(BLOCK_SIZE, device):
 
     @triton.jit
-    def add_with_mbarrier(
+    def tlx_square(
         x_ptr,
-        y_ptr,
         z_ptr,
         n_elements,
         BLOCK_SIZE: tl.constexpr,
     ):
         pid = tl.program_id(axis=0)
+        block_start = pid * BLOCK_SIZE
+
+        # create mbarrier for empty/full
+        bars_empty = tlx.alloc_barriers(num_barriers=1, arrive_count=BLOCK_SIZE)
+        bars_full = tlx.alloc_barriers(num_barriers=1, arrive_count=BLOCK_SIZE)
+
+        x_empty = tlx.local_view(bars_empty, 0)
+        x_full = tlx.local_view(bars_full, 0)
+
+        offsets = block_start + tl.arange(0, BLOCK_SIZE)
+        mask = offsets < n_elements
+
+        # Signal x_empty to wait
+        tlx.barrier_wait(bar=x_empty, phase=1)
+        # Signal x_full to expect 1 integer to be copied
+        tlx.barrier_expect_bytes(bar=x_full, size=4)
+
+        x = tl.load(x_ptr + offsets, mask=mask)
+
+        # Signal x_full to wait until it completes
+        tlx.barrier_wait(bar=x_full, phase=tl.constexpr(1))
```

Comment:
For a single warp-group program, let's test `barrier_wait` against `barrier_arrive` to make sure a they are working in pair. Something like,

```
tlx.barrier_arrive(bar, )
tlx.barrier_wait(bar, phase=0)
```